resources/dw/virtualwarehouse/impala/resource_impala_vw_acc_test.go

```diff
@@ -0,0 +1,152 @@
+	ClusterID         string
+	DatabaseCatalogID string
+	ImageVersion      string
+}
+
+func ImpalaPreCheck(t *testing.T) {
+	errMsg := "AWS CDW Impala Terraform acceptance testing requires environment variable %s to be set"
+	if _, ok := os.LookupEnv("CDW_CLUSTER_ID"); !ok {
+		t.Skipf(errMsg, "CDW_CLUSTER_ID")
+	}
+	if _, ok := os.LookupEnv("CDW_DATABASE_CATALOG_ID"); !ok {
+		t.Skipf(errMsg, "CDW_DATABASE_CATALOG_ID")
+	}
+}
+
+func TestAccImpala_basic(t *testing.T) {
+	params := impalaTestParameters{
+		Name:              cdpacctest.RandomShortWithPrefix(cdpacctest.ResourcePrefix),
+		ClusterID:         os.Getenv("CDW_CLUSTER_ID"),
+		DatabaseCatalogID: os.Getenv("CDW_DATABASE_CATALOG_ID"),
+	}
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			cdpacctest.PreCheck(t)
+			ImpalaPreCheck(t)
+		},
+		ProtoV6ProviderFactories: cdpacctest.TestAccProtoV6ProviderFactories,
+		CheckDestroy:             testCheckImpalaDestroy,
+		Steps: []resource.TestStep{
+			// Create and Read testing
+			{
+				Config: utils.Concat(
+					cdpacctest.TestAccCdpProviderConfig(),
+					testAccImpalaBasicConfig(params)),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("cdp_dw_vw_impala.test_impala", "name", params.Name),
+					resource.TestCheckResourceAttr("cdp_dw_vw_impala.test_impala", "cluster_id", params.ClusterID),
+					resource.TestCheckResourceAttr("cdp_dw_vw_impala.test_impala", "database_catalog_id", params.DatabaseCatalogID),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccImpalaBasicConfig(params impalaTestParameters) string {
+	return fmt.Sprintf(`
+		resource "cdp_dw_vw_impala" "test_impala" {
+		  cluster_id = %[1]q
+		  database_catalog_id = %[2]q
+		  name = %[3]q
+		  image_version = %[4]q
+		}
+	`, params.ClusterID, params.DatabaseCatalogID, params.Name, params.ImageVersion)
+}
+
+func testCheckImpalaDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "cdp_dw_vw_impala" {
+			continue
+		}
+
+		cdpClient := cdpacctest.GetCdpClientForAccTest()
+		params := operations.NewDescribeVwParamsWithContext(context.Background())
+		clusterID := rs.Primary.Attributes["cluster_id"]
+		params.WithInput(&models.DescribeVwRequest{
+			VwID:      &rs.Primary.ID,
+			ClusterID: &clusterID,
+		})
+
+		_, err := cdpClient.Dw.Operations.DescribeVw(params)
+		if err != nil {
+			if strings.Contains(err.Error(), "Virtual Warehouse not found") {
+				continue
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+// Set RUN_IMAGE_VERSION_TESTS to run this test
+func TestAccImpalaImageVersion(t *testing.T) {
+
+	OptionalPreCheck(t, "RUN_IMAGE_VERSION_TESTS")
+	params := impalaTestParameters{
+		Name:              cdpacctest.RandomShortWithPrefix(cdpacctest.ResourcePrefix),
+		ClusterID:         os.Getenv("CDW_CLUSTER_ID"),
+		DatabaseCatalogID: os.Getenv("CDW_DATABASE_CATALOG_ID"),
+		ImageVersion:      "2024.0.19.0-301",
```

Comment:
Get back based on the cluster id and the db catalog the available image list and select one from there and use that here. That way the test should be future proof

Reply:
Sure, sounds good.

Reply:
I tried this by using the DWClient for tests and have figured that there is no API call to List images . All the API calls are here

I tried to see if I can just directly hit the API endpoint but the first look does not show that hitting API endpoints is allowe

`	cdpClient := cdpacctest.GetCdpClientForAccTest()

	req, err := cdpClient.NewRequest("GET", fmt.Sprintf("/clusters/%s/versions", clusterID), nil)
	if err != nil {
		return "", fmt.Errorf("failed to create request: %w", err)
	}
`
I would work on it more to figure it out.

Reply:
Done !